pkg/pillar/cmd/zedagent/parseconfig.go

```diff
@@ -1574,6 +1574,81 @@ func parseU32ConfigItem(newGlobalConfigPtr *types.GlobalConfig, key string,
 
 var itemsPrevConfigHash []byte
 
+type ConfigItemType uint8
+
+const (
+	CONFIGITEM_TYPE_INT ConfigItemType = iota
+	CONFIGITEM_TYPE_BOOL
+	CONFIGITEM_TYPE_STRING
+	CONFIGITEM_TYPE_TRISTATE
+)
+
+type ConfigItemSpec struct {
+	Key      string
+	ItemType ConfigItemType
+
+	intMin     int32
+	intMax     int32
+	intDefault int32
+
+	stringDefault   string
+	boolDefault     bool
+	tristateDefault types.TriState
+}
+
+type ConfigItemSpecMap struct {
+	itemMap map[string]ConfigItemSpec
+}
+
+func (configMap ConfigItemSpecMap) AddIntItem(key string, defaultInt int32, min int32, max int32) {
+	log.Infof("Adding int item %s", key)
+	if defaultInt < min || defaultInt > max {
+		log.Infof("Adding int item %s failed", key)
+		return
+	}
+	configItem := ConfigItemSpec{
+		ItemType:   CONFIGITEM_TYPE_INT,
+		Key:        key,
+		intDefault: defaultInt,
+		intMin:     min,
+		intMax:     max,
+	}
+	configMap.itemMap[key] = configItem
```

Comment:
I think it makes sense to check that the key doesn't already exist in the map, and log.Fatalf in that case.
That applies to all the places where you are setting itemMap[key]